src/tools/sfdk/sdkmanager.cpp

```diff
@@ -1348,15 +1348,37 @@ bool SdkManager::mapEnginePaths(QString *program, QStringList *arguments, QStrin
         {cleanSharedSrc, Constants::BUILD_ENGINE_SHARED_SRC_MOUNT_POINT, caseInsensitiveOnWindows}
 #endif
     };
+    *program = QDir::fromNativeSeparators(*program);
+
     for (const Mapping &mapping : mappings) {
         qCDebug(sfdk) << "Mapping" << mapping.hostPath << "as" << mapping.enginePath;
         program->replace(mapping.hostPath, mapping.enginePath, mapping.cs);
-        arguments->replaceInStrings(mapping.hostPath, mapping.enginePath, mapping.cs);
+        QRegularExpression::PatternOption reCaseSensitivity = (mapping.cs == Qt::CaseInsensitive)
+                ? QRegularExpression::CaseInsensitiveOption : QRegularExpression::NoPatternOption;
+        QRegularExpression re(QString("(?:%1|%2)([^:;,'\"]*)")
+                              .arg(QRegularExpression::escape(mapping.hostPath))
+                              .arg(QRegularExpression::escape(QDir::toNativeSeparators(mapping.hostPath))),
+                              reCaseSensitivity);
+        auto map = [=](QString string) {
+            QRegularExpressionMatchIterator reIterator = re.globalMatch(string);
+            QMap<QString,QString> replacements;
```

Comment:
space around operators.

I would swap those two lines and add comment before, e.g.,`Replace longer matches first! Use map for sorting.`

Reply:
done